src/bugle_forms/model/form.clj

```diff
@@ -18,24 +21,58 @@
   {:form/id (UUID/randomUUID)
    :form/name name
    :form/owner user-id
+   :form/status :draft
    :form/created (Instant/now)})
 
 (defn insert!
   "Insert form in database."
   [form]
-  (sql/insert! db/datasource form-table form))
+  (->> (update form :form/status (comp jdbc-types/as-other name))
+       (sql/insert! db/datasource form-table)))
 
 (defn get
   "Retrieve form details."
   [id]
-  (if-some [form (sql/get-by-id db/datasource form-table id)]
+  (if-some [form (some-> (sql/get-by-id db/datasource form-table id)
+                         (update :form/status keyword))]
     form
-    {:error "Form not found."}))
+    {:error :form-not-found}))
+
+(defn link
+  "Get the share link for a form.
+  Returns `nil` if the form is invalid or unpublished."
+  [form]
+  (when (and (s/valid? ::specs/form form)
+             (= :published (:form/status form)))
+    (str "/form/" (:form/id form))))
 
 (defn get-forms
   "Retrieve forms belonging to a user."
   [{:keys [uuid]}]
-  (plan/select!
-   db/datasource
-   [:id :name :created]
-   ["select id, name, created from form where owner = ? order by created desc" uuid]))
+  (into []
+        (map (fn [row]
+               (-> (select-keys row [:form/id :form/name :form/owner
+                                     :form/status :form/created])
+                   (update :form/status keyword))))
+        (jdbc/plan db/datasource
+                   ["select id, name, owner, status, created
+                    from form where owner = ?
+                    order by created desc" uuid])))
+
+(defn published?
+  "Is the form having `id` published?"
+  [form]
+  (= :published (:form/status form)))
+
+(defn publish!
+  "Change form status to published.
+  Takes the user session data and form ID to publish."
+  [{user-id :uuid} id]
+  (let [form (bugle-forms.model.form/get id)]
```

Comment:
This doesn't seem to check for the case when the form ID doesn't exist. There isn't a test for this case either. We should probably check for this and return an appropriate error message.

Reply:
Will do

Reply:
btw, I have explicitly decided to show a vague error message to the end-user ("Something went wrong") regardless of the error on the backend. I believe you are referring to error messages for the user of this API, but I am mentioning this as well just in case.